spec.html

```diff
@@ -22277,6 +22277,195 @@ <h1>Example Source Text Module Record Graphs</h1>
+              </td>
+              <td>
+                The list of export bindings associated with this dynamic Module Record.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                [[EvaluationError]]
+              </td>
+              <td>
+                An abrupt completion | *undefined*
+              </td>
+              <td>
+                A completion of type ~throw~ representing the exception that occurred during evaluation.  *undefined* if no exception occurred or if [[Status]] is not `"evaluated"`.
+              </td>
+            </tr>
+            </tbody>
+          </table>
+        </emu-table>
+        
+        <emu-clause id="sec-createdynamicmodule" aoid="CreateDynamicModule">
+          <h1>CreateDynamicModule ( _realm_, _hostDefined_ )</h1>
+          <p>This method would be expected to be called by the host when constructing a Module Record in _HostResolveImportedModule_.</p>
+          <p>The abstract operation CreateDynamicModule with arguments _realm_, and _hostDefined_ creates a new Dynamic Module Record performing the following steps:</p>
+          <emu-alg>
+            1. Let _exportNames_ be a new empty List.
+            1. Return the Dynamic Module Record { [[Realm]]: _realm_, [[Environment]]: *undefined*, [[Namespace]]: *undefined*, [[Status]]: `"uninstantiated"`, [[ExportNames]]: _exportNames_, [[EvaluationError]]: *undefined*, [[HostDefined]]: _hostDefined_ }.
+          </emu-alg>
+        </emu-clause>
+
+        <p>The following definitions specify the required concrete methods for Dynamic Module Records.</p>
+
+        <emu-clause id="sec-getexportednames">
+          <h1>GetExportedNames ( _exportStarSet_ ) Concrete Method</h1>
+          <p>The GetExportedNames concrete method of a Dynamic Module Record implements the corresponding Module Record abstract method.</p>
+          <p>It performs the following steps:</p>
+          <emu-alg>
+            1. Let _module_ be this Dynamic Module Record.
+            1. Return _module_.[[exportNames]].
+          </emu-alg>
+        </emu-clause>
+  
+        <emu-clause id="sec-resolveexport">
+          <h1>ResolveExport ( _exportName_, _resolveSet_ ) Concrete Method</h1>
+          <p>The ResolveExport concrete method of a Dynamic Module Record implements the corresponding Module Record abstract method.</p>
+
+          <p>ResolveExport ensures that a binding is created for the dynamic Module Record, lazily creating a binding if needed.</p>
+
+          <p>This abstract method performs the following steps:</p>
+  
+          <emu-alg>
+            1. Let _module_ be this Dynamic Module Record.
+            1. Let _hasExistingExportBinding_ be set to *false*.
+            1. For each string _e_ in _module_.[[ExportNames]], do
+              1. If _e_ is equal to _exportName_ then,
+                1. Set _hasExistingExportBinding_ to *true*.
+            1. If _hasExistingExportBinding_ is *false* then,
```

Comment:
I wasn't sure if this whole approach was necessary, or if it is ok to just write:

```
1. If _module_.[[ExportNames]] does not contain an entry for _exportName_ then,
```

Reply:
yes, it is ok to do so. E.g.:

![image](https://user-images.githubusercontent.com/38969/42104114-6b215a06-7b99-11e8-9e94-7c363c0e1ad2.png)